duckduckgo/Android
# Retention Drivers Survey

Task/Issue URL: https://app.asana.com/0/1201807753394693/1204669492338873

### Description
Two new features for survey feature:

- We also notify the user about an available survey through a notification
- New 2 url parameters have been added: `da` and `src`

### Steps to test this PR
ℹ️ For all tests, in `SurveyService` change endpoint for survey() to `https://ddg-sandbox.s3.amazonaws.com/survey/v2/survey-mobile.json`
ℹ️ For changing survey's `installationDay` go to `App Inspector` and from the table `survey` edit `daysInstalled` value

_Survey notification_
- Fresh install
- When navigate to browser screen:
- [x] Check in-app cta is showing
- [x] Check you receive the notification 5 seconds later

_Access survey from notification_
- Fresh install
- When navigate to browser screen a notification will show in 5 seconds
- Tap on survey notification
- [x] Check it navigates you to Survey screen

_Finish Survey accessing from notification_
- Fresh install
- When navigate to browser screen a notification will show in 5 seconds
- Tap on survey notification
- Complete survey
- [ ] Check survey cta and notification are dismiss

_Finish Survey accessing from CTA_
- Fresh install
- When navigate to browser screen the survey cta will appear
- Tap on survey CTA
- Complete survey
- [ ] Check survey cta and notification are dismiss

_Survey _installationDay_ != 0_
- Fresh install
- Navigate to browser screen
- [ ] Check in-app cta is NOT showing
- [ ] Check you DON'T receive the notification 5 seconds later
- [ ] Check you receive the notification in day `installationDay` _(optional)_

_Survey _installationDay_ == -1 (survey targeting all users)_
- Fresh install
- Navigate to browser screen
- [ ] Check in-app cta is showing
- [ ] Check you receive the notification 5 seconds later

_Survey _installationDay_ == null (survey targeting users who installed the app >= 30 days)_
- Fresh install
- Navigate to browser screen
- **IF install day >= 30**
- [ ] Check in-app cta is showing
- [ ] Check you receive the notification 5 seconds later
- **IF install day < 30**
- [ ] Check in-app cta is NOT showing
- [ ] Check you DON'T receive the notification 5 seconds later

### No UI changes
